src/query/graphite/native/aggregation_functions.go

```diff
@@ -247,6 +247,55 @@ func groupByNode(ctx *common.Context, series singlePathSpec, node int, fname str
 		metaSeries[key] = append(metaSeries[key], s)
 	}
 
+	return applyFnToMetaSeries(ctx, series, metaSeries, fname)
+}
+
+// Takes a serieslist and maps a callback to subgroups within as defined by multiple nodes
+//
+//      &target=groupByNodes(ganglia.server*.*.cpu.load*,"sum",1,4)
+//
+// Would return multiple series which are each the result of applying the “sum” aggregation to groups joined on the
+// nodes’ list (0 indexed) resulting in a list of targets like
+//
+// 		sumSeries(ganglia.server1.*.cpu.load5),sumSeries(ganglia.server1.*.cpu.load10),sumSeries(ganglia.server1.*.cpu.load15),
+// 		sumSeries(ganglia.server2.*.cpu.load5),sumSeries(ganglia.server2.*.cpu.load10),sumSeries(ganglia.server2.*.cpu.load15),...
+//
+// NOTE: if len(nodes) = 0, aggregate all series into 1 series.
+func groupByNodes(ctx *common.Context, series singlePathSpec, fname string, nodes ...int) (ts.SeriesList, error) {
+	metaSeries := make(map[string][]*ts.Series)
+
+	nodeLen := len(nodes)
+	if nodeLen == 0 {
+		key := "all" // put into single group
```

Comment:
It seems that the "agg key" in the graphite-web version of this simply is empty string, giving a very strange response.
https://github.com/graphite-project/graphite-web/blob/master/webapp/graphite/render/functions.py#L4870
```py
def aggKey(series, nodes, pathExpression=None):
  # if series.name looks like it includes a function, use the first path expression
  if pathExpression is None and series.name[-1] == ')':
    pathExpression = _getFirstPathExpression(series.name)
  return '.'.join([getNodeOrTag(series, n, pathExpression) for n in nodes])
```

Perhaps we should just return an error here if `len(nodes) == 0` since it seems like it shouldn't return anything sensical (instead of special casing this to a special "all" group which behavior doesn't exist in the graphite-web version).

Reply:
Just to capture the offline discussion: 
1. when `len(nodes) == 0`, we aggregate the all series list into one. 
2. switch the key from "all" to "*" to be more graphite like.